### PR TITLE
menhir 20151112 and re-parallelize the build

### PR DESCRIPTION
**Pull request by ilovezfs**
- update to 20151112, which is required to build CompCert 2.6
- re-parallelize the build by isolating "stage2"

### CODE REVIEW

File: Library/Formula/menhir.rb [hunk 1]
```diff
@@ -13,7 +13,10 @@ class Menhir < Formula
   depends_on "ocaml"
 
   def install
-    ENV.deparallelize
+    # Workaround parallelized build failure by running stage2 on its own.
+    # Submitted to menhir-list@yquem.inria.fr on 24th Feb 2016.
+    system "make -C src library stage1"
```
Comment (bfontaine) [inline Library/Formula/menhir.rb]: Please use separate strings: `system "make", "-C", "src", "library", "stage1"`.

Comment (ilovezfs) [inline Library/Formula/menhir.rb]: That seems peculiar since there's no interpolation being done.

Comment (mikemcquaid) [inline Library/Formula/menhir.rb]: It stops a subshell being created unnecessarily.

Comment (ilovezfs) [inline Library/Formula/menhir.rb]: Thank you for explaining! I knew there must be some reason other than commaphilia.

**Comment by mikemcquaid** [comment]
10.9 build failing: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/42303/version=mavericks/testReport/junit/brew-test-bot/mavericks/install_menhir/

**Comment by ilovezfs** [comment]
Odd. OK let's see if it likes this better. PR refreshed.

**Comment by ilovezfs** [comment]
@mikemcquaid OK, that seems to have made it happy.

**Comment by apjanke** [comment]
WRT parallelization: It's a manually constructed `Makefile` and they probably just don't have the dependencies set up right and haven't tried building it in parallel. Check out `src/Makefile`.

```
bootstrap: .versioncheck stage1 stage2 stage3
...
stage1:
	@$(OCAMLBUILD) -build-dir _stage1 menhir.$(TARGET)
...
stage2:
	@$(OCAMLBUILD) -build-dir _stage2 -tag fancy_parser \
...
stage3:
	@$(OCAMLBUILD) -build-dir _stage3 -tag fancy_parser \
```

`bootstrap` depends on all the stages, but the stages don't depend on each other, so there's no sequencing between them. That might be a more natural place to set up the sequence dependencies; intermediate targets would work too.

**Comment by bfontaine** [comment]
cc @yurug

**Comment by mikemcquaid** [comment]
Thanks @ilovezfs!

**Pull request closed without merging**